biomaj_download/download/interface.py

```diff
@@ -254,7 +252,7 @@ def _set_retryer(self, stop_condition, wait_policy):
         if isinstance(stop_condition, tenacity.stop.stop_base):
             # Use the value directly
             stop_cond = stop_condition
-        elif isinstance(stop_condition, six.string_types):
+        elif isinstance(stop_condition, str):
```

Comment:
Str only or could also be bytes?

Reply:
Good question. Logically, this should be a string (this is text that we parse).

For direct usage of `biomaj_download` in python, I can't imagine why someone would use a `bytes` object.

For usage in `BioMAJ` we should think about:

1. how it is read from the config file ?
2. how it is passed by protocol buffers ?

For question 1, `ConfigParser` always read values as `str` and some tests ensure that this works.

For question 2, the value comes from the field `options`: from a protocol buffers point of view, this field is of type `map<string, string>` therefore the value passed to this method is of type `string`. According to [the doc](https://developers.google.com/protocol-buffers/docs/overview#scalar), this type is mapped to `str` in Python 3. So I'm rather confident for this.

Is it clear enough ?

Now, the `simpleeval` module seems happy with `bytes` so maybe it's just fine to also handle this case.

Reply:
L
So looks fine, let's merge with this, we will adapt later on if needed